### ISSUE TITLE
Instead of "Something went wrong..." display ssh error

### ISSUE THREAD

**Issue by koppor** [gitbutlerapp/gitbutler#5101]
### Version

0.13.0

### Operating System

Windows

### Distribution Method

msi (Windows)

### Describe the issue

When there are issues with git after quitting a conflict edit "Something went wrong..." is displayed.

### How to reproduce

1. I used GitButler
2. GitButler asked for resolving merge conflicts
3. I did not
4. I closed git butler
5. I closed my SSH key server
6. I restarted GitButler. 
7. GitButler said, I was in editing mode
8. I pressed "Cancel"
9. GitButler displayS: "Something went wrong..."

I checked the logs:

![image](https://github.com/user-attachments/assets/d0250420-5287-44ac-82a4-bac62d6096c8)

It seems that something is wrong with SSH

### Expected behavior

GitButler should output issues with git.

### Relevant log output

```shell
![image](https://github.com/user-attachments/assets/d0250420-5287-44ac-82a4-bac62d6096c8)
```

**Comment by Byron** [comment]
Thanks a lot for reporting!

A lot happened in the steps to reproduce, and some of that may very well be the reason for the failure.
However, the only idea I have relates to the SSH and keyserver parts. Now I wonder if SSH wanted to obtain a password, but couldn't do it as it needed a terminal? But then again it's unclear how it did ever work if restarting the keyserver is now a problem.

On MacOS with authentication and missing popups or environment variables can typically be resolved by launching GitButler from a terminal, but I don't know if that makes any difference on Windows.

**Comment by koppor** [comment]
> A lot happened in the steps to reproduce,

😇

> and some of that may very well be the reason for the failure. However, the only idea I have relates to the SSH and keyserver parts. Now I wonder if SSH wanted to obtain a password, but couldn't do it as it needed a terminal? 

It could not do it, because there was no server available. (The terminal itself worked IMHO)

> But then again it's unclear how it did ever work if restarting the keyserver is now a problem.

I was not clear: It is super simple: The server did not run. (Expanded: If users use [KeeAgent](https://lechnology.com/software/keeagent/#Download), the database for KeePass needs to run. And I did not run KeePass 😅)

> On MacOS with authentication and missing popups or environment variables can typically be resolved by launching GitButler from a terminal, but I don't know if that makes any difference on Windows.

Similar things here. (Long text on that at https://github.com/javakeyring/java-keyring?tab=readme-ov-file#security-concerns).

**Comment by Byron** [comment]
Thanks for the clarification :)! I think the real issue here is that while in this particular state, the UI ends up in `Something went wrong` mode, which is somewhat equivalent to a hard crash. It can be provoked by making Git fetches fail.

Ideally, the UI will gracefully handle this.

**Comment by koppor** [comment]
> Thanks for the clarification :)! I think the real issue here is that while in this particular state, the UI ends up in `Something went wrong` mode, which is somewhat equivalent to a hard crash. It can be provoked by making Git fetches fail.
> 
> Ideally, the UI will gracefully handle this.

Yeah - and it does at other places; it least shows up the error as notification.

**Comment by krlvi** [comment]
Thanks for reporting this. The screenshot shows two separate failures: an automatic fetch authentication error, and the UI trying to read workspace data while the repository was still in edit mode.

Both paths have since been fixed: #8542, released in `0.14.22`, keeps auto-fetch failures from taking down the project UI, and #13306, released in `0.19.8`, refreshes edit-mode state before navigating back to the workspace. Authentication failures now go through the normal warning UI, and #14858 further improves the message when Git can't obtain credentials or an SSH passphrase.

I'm closing this as fixed. If it still reproduces on `0.21.0` or newer, let us know and we'll reopen.
